modules/renter/contractor/downloader.go

```diff
@@ -79,7 +80,12 @@ func (hd *hostDownloader) Sector(root crypto.Hash) ([]byte, error) {
 
 	hd.contractor.mu.Lock()
 	hd.contractor.contracts[contract.ID] = contract
-	hd.contractor.saveSync()
+	cpath := fmt.Sprintf("contracts.%s", contract.ID.String())
+	hd.contractor.persist.update(
+		cpath+".lastrevision", contract.LastRevision,
+		cpath+".lastrevisiontxn", contract.LastRevisionTxn,
+		cpath+".downloadspending", contract.DownloadSpending,
+	)
```

Comment:
Is there any way we can be certain that this covers the whole spectrum of changes?

We're sort of changing paradigms here from something where you can expect all fields to be saved simultaneously to where you have to be explicit any time that you change something.

I know that it's verbose, but for safety reasons I think I'd prefer the updates to always cover all fields except the big ones.